File: microservice-gestion-cours/src/main/java/com/isi/model/Professeur.java
package com.isi.model;


import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.isi.model.base.Base;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import java.util.Collection;

@Entity
@Table(name = "isi_professeurs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Professeur extends Base {
    private Long id;
    private String nom;
    private String prenom;
    private String email;
    private String telephone;

    @JsonIgnore
    @OneToMany(mappedBy = "professeur")
    private Collection<Cours> cours;
}

File: frontenddevops/src/app/modules/professeur/models/professeur.ts
export interface Professeur {
    id: number;
    nom: string;
    prenom: string;
    email: string;
    telephone: string;
    departementId?: number;
    specialiteId?: number;
  }
  
  export interface ProfesseurDTO {
    nom: string;
    prenom: string;
    email: string;
    telephone: string;
    departementId?: number;
    specialiteId?: number;
  }